features/steps/thens/alignment.py

```diff
@@ -110,3 +127,43 @@ def step_impl(context, inst, ifc_rep_criteria, ifc_layout_criteria):
                                 yield ValidationOutcome(inst=inst, expected=ifc_layout_criteria,
                                                         observed="IfcAlignmentCant",
                                                         severity=OutcomeSeverity.ERROR)
+
+
+@gherkin_ifc.step(
+    'The alignment\'s {layout_type} layout must have the same number of segments as the shape representation')
+def step_impl(context, inst, layout_type):
+    for layout_ent in context.instances:
+        for rel in layout_ent.Nests:
+            ent = rel.RelatingObject
+            if ent.is_a() == "IfcAlignment":
+                align = ifc43x_alignment_validation.entities.Alignment().from_entity(ent)
+
+                match layout_type:
+                    case "horizontal":
+                        logic_count, rep_count = count_segments(
+                            logic=align.horizontal,
+                            representation=align.composite_curve,
+                        )
+                    case "vertical":
+                        logic_count, rep_count = count_segments(
+                            logic=align.vertical,
+                            representation=align.gradient_curve,
+                        )
+                    case "cant":
+                        logic_count, rep_count = count_segments(
+                            logic=align.cant,
+                            representation=align.segmented_reference_curve,
+                        )
+                    case _:
+                        msg = f"Invalid value '{layout_type}'.  Should be 'horizontal', 'vertical', or 'cant'."
+                        raise NameError(msg)
+
+                if logic_count != rep_count:
+                    # account for alignment that has business logic only
+                    # or contains representation only
+                    logic_only = (logic_count is not None) and (rep_count is None)
+                    rep_only = (logic_count is None) and (rep_count is not None)
+                    if not (logic_only or rep_only):
+                        yield ValidationOutcome(inst=inst, expected="same count of segments",
+                                                observed="differing count of segments",
+                                                severity=OutcomeSeverity.ERROR)
```

Comment:
I'd say just add `logic_count` and `rep_count` instead of the string that is currently parsed to `expected` and `observed`
Additionally with the feature file 'same count' it should be possible for the end-user to derive what the error is. 

Or what do you think? I'm not completely sure

Reply:
I updated `observed` to report the number of segments found in the business logic and in the shape representation.  Example
```
Observed value: {"value": "10 segments in business logic and 8 segments in representation"}
```